deepspeed/inference/engine.py

```diff
@@ -432,16 +455,18 @@ def _load_checkpoint(self, load_dir, load_module_strict=True, tag=None):
         else:
             sd_loader = SDLoaderFactory.get_sd_loader_json(load_dir, self.checkpoint_engine)
 
-        if type(sd_loader) is list:
-            self.sd = torch.load(sd_loader[0], map_location='cpu')
+        checkpoint = sd_loader['checkpoints']
```

Comment:
Is this backwards-compatible? It seems that `sd_loader` has changed?

Reply:
previously, the autoTP checkpoint path could not work with KI checkpoint format. KI checkout point is like "{"type": "ds_model", "checkpoints": ["/skyrex01/wangyi/gpt-j6/hub/models--EleutherAI--gpt-j-6B/snapshots/f98c709453c9402b1309b032f40df1c10ad481a2/pytorch_model.bin"], "version": 1.0}
"

Reply:
I see. I will also leave this for KI experts to review.